### PR TITLE
Migrate ERC20

### PR DESCRIPTION
**Pull request by andrew-fleming**
WIP

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.SILENT:
+.PHONY: compile
+SOURCE_FOLDER=./src/openzeppelin
+
+install:
+	git submodule init && git submodule update && cp -rf cairo/corelib .
+
+update:
+	git submodule update && cp -rf cairo/corelib .
```
Comment (Amxx) [inline Makefile:9]: Why no do 
```suggestion
install: init update
init:
	git submodule init
update:
	git submodule update
	cp -rf cairo/corelib .
```
and avoid code dupliation

File: src/openzeppelin/token/erc20/erc20.cairo [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    ///
+    /// Internals
+    ///
+
+    fn initializer(name_: felt, symbol_: felt, initial_supply: u256, recipient: ContractAddress) {
+        _name::write(name_);
+        _symbol::write(symbol_);
+        _mint(recipient, initial_supply);
```
Comment (Amxx) [inline src/openzeppelin/token/erc20/erc20.cairo:135]: what is the rational for including a `_mint` into the initializer ?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.SILENT:
+.PHONY: compile
+SOURCE_FOLDER=./src/openzeppelin
+
+install:
+	git submodule init && git submodule update && cp -rf cairo/corelib .
```
Comment (Amxx) [inline Makefile:6]: `corelib` should be added to the `.gitignore`.

**Comment by martriay** [comment]
i deleted the `cairo-1.0` branch before re-targeting this PR to `cairo-1`, which closed this PR. my bad! will reopen.

**Pull request closed without merging**